common/utils/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -145,6 +145,30 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
     return size.get();
   }
 
+  public static void cleanDirectory(File dir) throws IOException {
+    if (dir == null || !dir.exists() || !dir.isDirectory()) {
+      throw new IllegalArgumentException("Invald input directory " + dir.getAbsolutePath());
+    }
+    cleanDirectory(dir.toPath());
+  }
+
+  public static void cleanDirectory(Path rootDir) throws IOException {
+    Files.walkFileTree(rootDir, new SimpleFileVisitor<Path>() {
```

Comment:
Although there is exception input validation when using `cleanDirectory(File dir)` as the entry point, since this method is also public, should it also perform exception input validation?

Reply:
Thanks. For that concern, I'll make this `private` properly at next commit.